cdap-docs/admin-manual/source/operations/logging.rst

```diff
@@ -50,28 +102,409 @@ CDAP supports logging through standard `SLF4J (Simple Logging Facade for Java)
+
+.. figure:: /_images/logging/logging-framework.png
+    :figwidth: 100%
+    :width: 800px
+    :align: center
+
+    **CDAP Logging Framework:** From YARN containers, through Kafka and the Log Saver
+    Service, to HDFS
+
+Logging Standards
+-----------------
+Logging uses the standard `SLF4J (Simple Logging Facade for Java)
+<http://www.slf4j.org/manual.html>`__ APIs and `Logback
+<https://logback.qos.ch/manual>`__. Logging is configured using instances of Logback's
+"logback" file, consisting of *logging pipelines* with *logging appenders*:
+
+  - A **logging pipeline** is a process that consumes log events from Kafka, invokes
+    *logback appenders* defined in its configuration, buffering, sorting, and then
+    persisting the log messages.
+
+  - A **logback appender** (or *appender*) is a Java class, responsible for writing the
+    log events to persistent storage and maintaining metadata about that storage.
+
+Configuration Files
+-------------------
+- The ``logback-container.xml`` configures the logs of YARN containers, both locally and to Kafka,
+  for application programs and any :ref:`CDAP System Services
+  <logging-monitoring-system-services-logs>` (such as the *Metrics Service*) that run in YARN. 
+- The ``logback.xml`` configures the edge node-located :ref:`CDAP System Services
+  <logging-monitoring-system-services-logs>`, such as the *App Fabric* or *Data Service*.
+- Both of these files are, in Distributed CDAP, located in ``/etc/cdap/conf``.
+- In the case of :ref:`Standalone CDAP <modes-data-application-platform>`, as containers
+  are replaced with Java threads, only a single file (``logback.xml``, located in
+  ``<cdap-sdk-home>/conf``) is used, replacing the ``logback.xml`` and
+  ``logback-container.xml`` of Distributed CDAP.
+- The ``cdap-site.xml`` controls the CDAP Log Pipeline for CDAP System Services, and the
+  reading, buffering, and memory allocated.
+- Custom "logback" files, if present in a specified directory, trigger the creation of
+  custom logging pipelines, with a pipeline created for each custom file.
+
+.. _logging-monitoring-custom-logging-pipeline:
+
+Custom Logging Pipeline
+=======================
+For a custom logging pipeline, you would create and configure a Logback XML file,
+configuring loggers, appenders, and properties based on your requirements, and place the
+file at the path specified in the ``cdap-site.xml`` file by the property
+``log.process.pipeline.config.dir`` of the ``cdap-site.xml`` file.
+
+Each custom pipeline requires a unique name, which is used for persisting the data and the
+retrieving of metadata. Properties controlling the pipeline (the
+``log.process.pipeline.*`` properties) are described :ref:`above
+<custom-log-pipeline-logging-configuration>`.
+
+For every XML file in the ``log.process.pipeline.config.dir`` directory, a separate logging pipeline is created, providing
+isolation from other logging pipelines. As they are separate Kafka consumers and
```

Comment:
Does this isolation mean that they run as separate processes? If one crashes, the other one survives? Or what exactly is this isolation?